cmd/cstor-pool-mgmt/controller/pool-controller/handler.go

```diff
@@ -525,3 +536,38 @@ func (c *CStorPoolController) getDeviceIDs(csp *apis.CStorPool) ([]string, error
 	// TODO: Add error handling
 	return pool.GetDeviceIDs(csp)
 }
+
+func (c *CStorPoolController) upgrade(csp *apis.CStorPool) (*apis.CStorPool, error) {
```

Comment:
can these two be functions rather than methods?

Reply:
To update the objects the functions will require the `clientset` present in the `CStorPoolController` object. The methods can be made into functions if we update the upgraded object in the sync function.

Reply:
ok.. we are using clientset of controller